testing/utils.py

```diff
@@ -0,0 +1,52 @@
+import os
+from contextlib import contextmanager
+from collections import Mapping
+
+
+@contextmanager
+def cd(path):
+    """
+    A context manager that temporarily changes the current working directory
+    """
+    old_dir = os.getcwd()
+    os.chdir(path)
+    try:
+        yield
+    finally:
+        os.chdir(old_dir)
+
+
+def deep_update(orig, override):
```

Comment:
So the motivation for this method is because the configs are stored at the class-level?

Reply:
It would just be used to update a default configuration with a custom one. If the configuration values fitted in a simple ``dict``, we could just use ``dict.update``, but here we have one more level (with the ``exhale_args`` attribute).

It was just a quick way to do it, as I had this ``deep_update`` function in my toolbox, so feel free to change the way it's done if you find a better / clearer way!